Assets/Scripts/Spells/Abstract/ShootingSpell.cs

```diff
@@ -0,0 +1,28 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using UnityEngine;
+
+namespace Assets.Scripts.Spells
+{
+    public abstract class ShootingSpell : Spell
+    {
+        protected GameObject WizardGameObj;
+
+        public override void Activate()
+        {
+            TurnManager turnManager = GameObject.FindObjectOfType<TurnManager>();
```

Comment:
Troche dziwne wydaje mi sie to by Szpel mial szukac samemu wlasciela i sie samemu wystrzeliwac.

Reply:
Hmmm wydaje mi się, że tak jak jest ok :).
Spell jest wystrzeliwany z kontrolera , tam jest wywoływana metoda Active na zaklęciu.
Spelle mogą mieć różne zachowania (mogą korzystać z pozycji jednego z czarodziejów/ globalny/ wybór kursorem). Także jeśli miałbym to przenieść do klasy wizardController to musiałbym rozróżniać w zależności od typu czaru.